test/plugin_functional/plugins/core_app_status/public/plugin.tsx

```diff
@@ -30,27 +37,37 @@ export class CoreAppStatusPlugin
       title: 'App Status',
       euiIconType: 'snowflake',
       updater$: this.appUpdater,
-      async mount(context, params) {
+      async mount(params: AppMountParameters) {
         const { renderApp } = await import('./application');
-        return renderApp(context, params);
+        return renderApp('app_status', params);
+      },
+    });
+    core.application.register({
+      id: 'app_status_start',
+      title: 'App Status Start Page',
+      updater$: this.appUpdater,
+      async mount(params: AppMountParameters) {
+        const { renderApp } = await import('./application');
+        return renderApp('app_status_start', params);
       },
     });
 
     return {};
   }
 
   public start(core: CoreStart) {
-    return {
+    const startContract = {
       setAppStatus: (status: Partial<AppUpdatableFields>) => {
         this.appUpdater.next(() => status);
       },
       navigateToApp: async (appId: string) => {
         return core.application.navigateToApp(appId);
       },
     };
+    window.__coreAppStatus = startContract;
+    return startContract;
```

Comment:
I need to be able to test NP to NP app navigation. As we are not loading LP plugins in non-legacy mode, I can't use the `core_provider_plugin` plugin anymore and I'm forced to expose my own API to the global scope.

Reply:
You could add buttons to click on instead, but this is probably less flaky.

Reply:
for `navigateToApp` I could (even if for core FT I think I prefer using directly our APIs), but not for the `setAppStatus` I'm also using